docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+    "nonce": "fsTLlOIUqtJHomMB2t6HymoAqJi-wORIFtg3y8c65VY",
+    "email": "alice@example.com"},
+"signatures":[
+  {
+    "protected":{
+      "alg": "RS256",
+      "kid": "1234",
+      "typ": "JWT"
+    },
+    "signature": "GqjU... (Issuer's signature)"
+  },
+  {
+    "protected": {
+    "alg": "ES256",
+    "typ": "CIC",
+    "rz": "b9522b5c4cff90687ec6787236184659e077a619b82827227114108440fec26a",
+    "upk": {
+        "alg": "ES256",
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "cvqyUFNs1OUdRcDSmzJfS7ynuTHAjlDqoeinCZy_r1Q",
+        "y": "Whl5jJUIz7ujFvlB5Hzhaz6DIlpyWQmIIA3J7VMj53o"
+    }},
+    "signature":"TBPZa...  (Clients's signature)"
+  },
+  {
+    "protected":{
+      "alg": "ES256",
+      "auth_time": 1722113589,
+      "eid": "6e38311685191ac62e8647e8263f98ad1d57752dab14a5b83298c0d70fe19942",
+      "exp": 1722117189,
+      "iat": 1722113589,
+      "iss": "https://mfa-cosigner.openpubkey.com",
+      "kid": "7890",
+      "nonce": "a958dd0574393cc3fa0e423b4d060009b44ecc710d1ff7af0e4cc74b9fc99649",
+      "ruri": "http://localhost:54435/mfacallback",
+      "typ": "COS"
+    },
+    "signature":"hiyDh... (Cosigner's signature)"
+  }
+]}
+```
+
+### Protected Header Claims
+
+Protected headers in JWS [(RFC-7515 Section 2)](https://datatracker.ietf.org/doc/html/rfc7515#section-2) are used in JWS as a place to put signature header parameters. That is metadata about the signature necessary to verify the signature. For instance parameters such as, `alg` which specifies the algorithm used to generate the signature or `kid` which is the Key ID of the public key used to verify the signature.  In PK Tokens we use protected headers in a new way namely to store claims the signature is making about the identity. Next, we will look at why we need to use protected headers in this new way.
+
+In an ID Token or JWT, the claims the issuer is making are set in the payload. However this isn't sufficient for OpenPubkey, signing parties other than the issuer may wish to add additional claims along with their signature. For instance a signing party who independently authenticated the identity, might want to add an additional claim specifying the time at which this independent authentication took place. This party can not update the payload without breaking the OP's signature on the payload.Our solution is to allow signing parties to specify additional claims in the protected header that signing party creates. This enables the signing party to add claims without requiring any new signatures from any other parties.
+
+#### Required Claims
+
+In PK Tokens the following claims are required and assumed to exist `alg`, `typ` and `kid`. The one exception is that since OP (OpenID Providers) operate outside the OpenPubkey protocol they can generate an OP signature and OP protected header that may not include `typ` or `kid`. However in all other signatures in a PK Token, OpenPubkey can enforce that `alg`, `typ` and `kid` are specified.
```

Comment:
I'm not sure I'm following what we mean here. If OPK is built off of existing compatibility with OIDC providers, how can we assume that `typ` and `kid` exist if OIDC providers operate outside of that sphere? Do we have additional detail here on which OPs don't support these claims and what they use instead?

Reply:
We can't assume they exist, so we need to be careful and handle the case where the OP doesn't. Let me reword.